### PR TITLE
RHDM-1145: [DMN Designer] Java Date field is not converted to DMN date type

### PR DESCRIPTION
**Pull request by danielzhe**
RHDM-1145: [DMN Designer] Java Date field is not converted to DMN date type
DROOLS-4977: [DMN Designer] Refactor DataObjectsServiceImpl to not duplicate logic from ParseHelper

EDIT: is not a part of an ensemble anymore. The changes in the Drools was superseded by https://github.com/kiegroup/drools/pull/2735

### CODE REVIEW

File: kie-wb-common-dmn/kie-wb-common-dmn-backend/src/main/java/org/kie/workbench/common/dmn/backend/editors/types/DataObjectsServiceImpl.java [hunk 1]
```diff
@@ -152,24 +148,32 @@ private String convertDataType(final String typeName,
...
-            return BuiltInType.DATE;
-        } else if (LocalTime.class.isAssignableFrom(clazz) || OffsetTime.class.isAssignableFrom(clazz)) {
-            return BuiltInType.TIME;
-        } else if (ZonedDateTime.class.isAssignableFrom(clazz) || OffsetDateTime.class.isAssignableFrom(clazz) || LocalDateTime.class.isAssignableFrom(clazz)) {
-            return BuiltInType.DATE_TIME;
-        } else if (Boolean.class.isAssignableFrom(clazz)) {
-            return BuiltInType.BOOLEAN;
-        } else if (Map.class.isAssignableFrom(clazz)) {
-            return BuiltInType.CONTEXT;
+        final Type type = ParserHelper.determineTypeFromClass(clazz);
```
Comment (manstis) [inline kie-wb-common-dmn/kie-wb-common-dmn-backend/src/main/java/org/kie/workbench/common/dmn/backend/editors/types/DataObjectsServiceImpl.java]: Looks like you'll need to change this to use `JavaBackedType.determineTypeFromClass(..)`. See [here](https://github.com/kiegroup/drools/pull/2735/files#diff-53384294028099fe193f10e79b1daab7R139).

**Comment by danielzhe** [comment]
@jomarko @manstis Changes applied here. The other PR was closed because it was superseded by https://github.com/kiegroup/drools/pull/2735.

**Comment by danielzhe** [comment]
@jomarko Ready for your review when you can.

**Comment by danielzhe** [comment]
@manstis Can we merge, please?

**Pull request merged**